cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -826,31 +826,31 @@ def exec_module(self, **kwargs):
                                     )
                     self.log("Update virtual machine with parameters:")
                     self.log(self.serialize_obj(vm_resource, 'VirtualMachine'), pretty_print=True)
-                    self.results['state'] = self.create_or_update_vm(vm_resource)
+                    self.results['ansible_facts']['azure_rm_vm'] = self.create_or_update_vm(vm_resource)
 
                 # Make sure we leave the machine in requested power state
                 if powerstate_change == 'poweron' and self.results['state']['powerstate'] != 'running':
                     # Attempt to power on the machine
                     self.power_on_vm()
-                    self.results['state'] = self.serialize_vm(self.get_vm())
+                    self.results['ansible_facts']['azure_rm_vm'] = self.serialize_vm(self.get_vm())
 
                 elif powerstate_change == 'poweroff' and self.results['state']['powerstate'] == 'running':
                     # Attempt to power off the machine
                     self.power_off_vm()
-                    self.results['state'] = self.serialize_vm(self.get_vm())
+                    self.results['ansible_facts']['azure_rm_vm'] = self.serialize_vm(self.get_vm())
 
                 elif powerstate_change == 'restarted':
                     self.restart_vm()
-                    self.results['state'] = self.serialize_vm(self.get_vm())
+                    self.results['ansible_facts']['azure_rm_vm'] = self.serialize_vm(self.get_vm())
 
                 elif powerstate_change == 'deallocated':
                     self.deallocate_vm()
-                    self.results['state'] = self.serialize_vm(self.get_vm())
+                    self.results['ansible_facts']['azure_rm_vm'] = self.serialize_vm(self.get_vm())
```

Comment:
it would be easier to just 'outdent' this instead of repeating the same line over and over